### DISCUSSION TITLE
Calling rust code from guile

### DISCUSSION THREAD

**Discussion by Immortalin** [guile-rs/guile-rs#17] [Q&A]
How do you call rust code from guile? (Like in the turtle tutorial)

**Comment by lidavidm** [comment]
Sorry I didn't see this earlier! I don't believe that's been implemented - this was really just a project to try and learn a bit about FFI in Rust. If I pick this back up again I'll look into it.

**Comment by aryalaadi** [comment]
create a `dylib` crate, with a C abi function which you will use to init the scheme module
you can define scheme stuff with functions like `scm_c_define_gsubr` [defined here](https://github.com/guile-rs/guile-rs/blob/fdbaef6acc0e45db5d9caf5d2a31eea657c1ba96/guile-sys/src/bindings.rs#L17010)
```scheme 
(define yourlib (dynamic-link "./yourlib.so"))
(dynamic-call "the_init_function" yourlib)
; and with this you can call rust code from guile
```

**Reply by Immortalin** [reply]
That's more of a hack than a proper two way binding though

**Reply by Immortalin** [reply]
This is what I meant by the tortoise tutorial
https://www.gnu.org/software/guile/docs/guile-tut/tutorial.html#Tortoise
